File: io/json/JsonException.java
package common.io.json;

import com.google.gson.JsonElement;

public class JsonException extends Exception {

	public static enum Type {
		TYPE_MISMATCH, TAG, UNEXPECTED_NULL, FUNC, INTERNAL, UNDEFINED
	}

	private static final long serialVersionUID = 6451473277106188516L;

	public JsonException(Type type, JsonElement elem, String str) {
		super(str);
		// TODO
	}

}

File: io/json/JsonException.kt
package common.io.json

import com.google.gson.JsonElement
import common.io.json.JsonExceptionimport

class JsonException(type: JsonException.Type?, elem: JsonElement?, str: String?) : Exception(str) {
    enum class Type {
        TYPE_MISMATCH, TAG, UNEXPECTED_NULL, FUNC, INTERNAL, UNDEFINED
    }

    companion object {
        private const val serialVersionUID = 6451473277106188516L
    }
}
